pkg/mutation/assignmeta_mutator.go

```diff
@@ -123,3 +124,29 @@ func isMetadataPath(path *parser.Path) bool {
 	}
 	return false
 }
+
+// IsValidAssignMetadata returns an error if the given assignmetadata object is not
+// semantically valid
+func IsValidAssignMetadata(assignMeta *mutationsv1alpha1.AssignMetadata) error {
+	path, err := parser.Parse(assignMeta.Spec.Location)
+	if err != nil {
+		return errors.Wrapf(err, "Invalid location format")
+	}
+	if !isMetadataPath(path) {
```

Comment:
nit: should this be `isValidMetadataPath()` to indicate that it's not just testing whether the path has a metadata root?

Reply:
Done